Updating yaml files with the new policy definition for system mounts

denisonbarbosa opened the pull request.

denisonbarbosa committed:
Updating yaml files with the new policy definition for system mounts

denisonbarbosa committed:
Refresh policy definition files

didrocks commented on cmd/admxgen/defs/mount.yaml:
```diff
@@ -17,3 +17,25 @@
+  displayname: "System mounts"
+  explaintext: |
+    Define network shares that will be mounted for the system.
+    If more shares are defined higher in other GPO, the entries listed here will be appended to the list and duplicates will be removed.
```
`higher in the GPO hierarchy`? (if we are not consistent with other descriptions, we should)

didrocks commented on cmd/admxgen/defs/mount.yaml:
```diff
@@ -17,3 +17,25 @@
+    Examples:
+      nfs://example.com/nfs_share
+      smb://example.com/smb_share
+      ftp://example.com/ftp_share
```
ftp can be user@password ? As there is no kerberos tickets for those mounts, I think we should for all protocols, provides the user / password basic authentication scheme in the example.

denisonbarbosa replied:
Well... This complicate things a lot. We don't account for this in the policy manager (nor anywhere else in the code, tbh). And, as those values will be stored in the system in a file (take the user mounts, for example), shouldn't we avoid having this kind of authentication for safety reasons?

didrocks replied:
So, what’s the difference for system mount between:
`nfs://example.com/nfs_share` and `[anonymous]nfs://example.com/nfs_share`?

denisonbarbosa replied:
The options that we define on the unit file, basically...

didrocks replied:
Can you be a little bit more precise? (and how does it effect a mount where we don’t pass username/password?)

denisonbarbosa replied:
Well, from what I've read, system mounts do not have a specific "anonymous" option for them. Which means I'll need to set some other options to specify that the user that is interacting with the mounted location shouldn't be checked and so on...

didrocks reviewed (changes requested):
Some comments suggestion

didrocks commented on cmd/admxgen/defs/mount.yaml:
```diff
@@ -17,3 +17,25 @@
+      ftp://example.com/ftp_share
+
+    Anonymous mounts should be prefixed by the [anonymous] tag:
+      [anonymous]<protocol>://<hostname-or-ip>/shared-dir
```
Note: this needs to be updated with the removal of [anynoumous] tag and telling that only anynomous mount are supported for system mounts.

denisonbarbosa committed:
Addressing PR comments

denisonbarbosa committed:
Refresh policy definition files

denisonbarbosa committed:
Updating definitions to show protocols supported by the mount command.
The examples were listed as smb and ftp, which are not types directly
supported by the mount command. Instead, it's now showing cifs and fuse
(even though we convert the previous examples, it's better to show the
actual value).

denisonbarbosa committed:
Refresh policy definition files

denisonbarbosa commented on cmd/admxgen/defs/mount.yaml:
```diff
@@ -26,3 +26,32 @@
+
+    This pattern must be followed, otherwise the policy will not be applied.
+    
+    By the default, the mounts will be done in anonymous mode. In case of authentication needed, a krb5 tag can be added to the value, e.g.
```
Did you share the folder anonymously?

denisonbarbosa replied:
IIRC NFS has a different approach than anonymous per se... It has something called "access based enumeration" which basically translates to: "everyone can mount this, but they will only see and do what they are allowed to"

denisonbarbosa commented on cmd/admxgen/defs/mount.yaml:
```diff
@@ -26,3 +26,32 @@
+        <protocol>://<hostname-or-ip>/<shared-dir>
+    e.g.
+        nfs://example_nfs.com/nfs_shared_dir
+        cifs://example_cifs.com/cifs_shared_dir
+        fuse://ftp_share_server.com
```
We accept [both](https://github.com/ubuntu/adsys/blob/f6cf6e2d06391a646651201533b2dc5dd836ba84/internal/policies/mount/mount.go#L310) 😄.

However, as I've said in the commit rationale, we should list the correct input in the policy to avoid creating some "laziness"

denisonbarbosa replied:
For the user is different... Gio supports the smb:// directly, so there's no need to do it.

denisonbarbosa replied:
After what we talked in IIRC, about gio having to use smb:// instead of cifs:// and so on, do you think we can merge this?

denisonbarbosa committed:
Changing system documentation to be in line with the user one.

denisonbarbosa committed:
Update cmd/admxgen/defs/mount.yaml

Co-authored-by: Gabriel Nagy <gabriel.nagy@canonical.com>

github-actions[bot] committed:
Refresh policy definition files (#545)

Co-authored-by: denisonbarbosa <denisonbarbosa@users.noreply.github.com>
